core/web/helpers.go

```diff
@@ -28,6 +28,19 @@ func jsonAPIError(c *gin.Context, statusCode int, err error) {
 	}
 }
 
+func jsonAPIErrorNotFound(c *gin.Context, err error) {
+	jsonAPIError(c, http.StatusNotFound, err)
```

Comment:
I'll likely have to toggle these to 500s given it may be a couple weeks before we understand whether this will break anyone.

Reply:
I don't have a strong opinion on how to manage this switch, but just as a nit do these helper funcs add value? They save a few characters compared to inline, but they are harder to read and hide the interesting bit.

Reply:
But why is it better to call an obscure helper function `jsonAPIErrorNotFound(c, err)` rather than do an inline `jsonAPIError(c, http.StatusNotFound, err)`? Are there other benefits besides saving a few characters?

Reply:
I had to click through `jsonAPIError` to understand its behavior when updating these APIs. Now there's a second hop, but the definition for `jsonAPIError` is right above so perhaps it's not too bad?

The benefit I see is that I don't want future development to think about what http status codes to use, just to reason about what type of negative path is being observed (i.e. bad request, resource not found, or unauthorized). Then using these helpers gives you the http status code for free. If I had no knowledge of this change and were to do this unit of work again, I would be faced with the decision "do I throw a 500 for a [bad request, resource not found] or the proper status code?" and I want to remove that line of questioning.

There might be a cleaner or more idiomatic way to do this that I'm missing - any ideas?

Reply:
> The benefit I see is that I don't want future development to think about what http status codes to use, just to reason about what type of negative path is being observed (i.e. bad request, resource not found, or unauthorized

But if those are 1-1 with the status codes and the code name is repeated inside of the function name, then how is it simpler?

I can see that we are forwarding now, which aids the transition from one code to another, but I think that further highlights the reduction in readability - now devs reading those calls are going to mistakenly believe that one code is being returned when the helper actually does something else entirely.

Reply:
> But if those are 1-1 with the status codes and the code name is repeated inside of the function name, then how is it simpler?

The effect I'm going for here is to make it harder to use the wrong status code for a standard negative path. You always have to reason about what the issue is, but choosing the correct status code is a decision that comes after. Based on our discussion, it seems the tradeoff is equal or less readability.

> I can see that we are forwarding now, which aids the transition from one code to another, but I think that further highlights the reduction in readability - now devs reading those calls are going to mistakenly believe that one code is being returned when the helper actually does something else entirely.

Definitely, good point. This state should only last a few days - we should be good to use the correct status codes for a few reasons, but just want to be cautious until cleared. Once we are cleared, their assumptions would live up with reality :)